desc/coils.py

```diff
@@ -238,6 +252,7 @@ def current(self):
 
     @current.setter
     def current(self, new):
+        new = setdefault(new, 1, new)
```

Comment:
What is the purpose of this? Wouldn't this prevent someone from setting the current to 0, since 0 = False and so it would get replaced with the default value of 1?

Reply:
I think you might have meant to do `new = setdefault(new, 1)`

Reply:
check if can set current to 0